fix postmap exec to update *.db files if source is newer or missing

georgehansper opened the pull request:
The exec parameter `creates` inhibits the exec from running after the nominated file is has been created, even when the exec is being 'notified'.

https://puppet.com/docs/puppet/6.4/type.html#exec

>    If the exec would not have run (due to an `onlyif`, `unless`, or `creates` attribute) and receives an event, it still will not run.

This `unless` clause will run the exec whenever:
* the db file is older than the text file
* the db file is missing

Commit:
fix postmap exec to update *.db files if source is newer or missing

trefzer commented:
Hi
You're right, thanks a lot for the patch.
I only have one problem.... on Debian stretch the test binary is under /usr/bin/test !
Could you change the patch to add a search path to the exec and use test with an unqualified path ?

Thanks
Benedikt

georgehansper commented:
That's what I had originally, but I thought it was giving me an error, and yes, it does:
```
      Failure/Error: it { is_expected.to compile.with_all_deps }
        error during compilation: Parameter unless failed on Exec[rebuild map mymap]: 'test /etc/postfix/maps/mymap.db -nt /etc/postfix/maps/mymap' is not qualified and no path was specified. Please qualify the command or specify a path. (file: /builds/mirrors/postfix/spec/fixtures/modules/postfix/manifests/map.pp, line: 75)
```

I'll add the `path` attribute.

georgehansper commented:
I've also changed:
```
      subscribe   => Concat[$filename],
```
to
```
      require => Concat[$filename],
```

I expect that they would do the same thing, but `subscribe` only makes sense if `refreshonly` is being used (which I deleted). `require` is more explicit in this case.

georgehansper commented:
I've also changed:
```
      subscribe   => Concat[$filename],
```
to
```
      require => Concat[$filename],
```

I expect that they would do the same thing, but `subscribe` only makes sense if `refreshonly` is being used.

trefzer commented:
thanks a lot, now it looks good, will merge now.

georgehansper commented:
Thanks, will do.
